pkg/cmd/server/api/types.go

```diff
@@ -420,6 +420,8 @@ type MasterClients struct {
 }
 
 type DNSConfig struct {
+	// DNSDomain is the authoritative domain of the DNS server.
+	DNSDomain string
```

Comment:
Need to add some validation around this to ensure it is present (defaulting should ensure it is set to cluster.local) and that it has the expected format (there are existing validation functions for validation DNS names)